tests/test_image.py

```diff
@@ -0,0 +1,221 @@
+    return func
+
+
+@pytest.fixture
+def mock_session_method(session, mocker):
+    """Return a helper function to mock the return value(s) of a session method using a simple syntax."""
+    def func(method_name, return_values):
+        mocker.patch.object(session, method_name, side_effect=return_values)
+    return func
+
+# TESTS
+
+# DOCSTRINGS
+
+
+def test_image_class_has_docstring():
+    assert Image.__doc__ is not None
+
+
+def find_members(*classes, member_type=types.FunctionType):
+    for clazz in classes:
+        for name in dir(clazz):
+            if not name.startswith('__') and isinstance(getattr(clazz, name), member_type):
+                yield getattr(clazz, name)
+
+
+@pytest.mark.parametrize("member", find_members(Image))
+def test_image_methods_have_docstrings(member):
+    assert member.__doc__ is not None
+
+
+@pytest.mark.parametrize("member", find_members(Image, member_type=types.MethodType))
+def test_image_classmethods_have_docstrings(member):
+    assert member.__doc__ is not None
+
+
+@pytest.mark.parametrize("member", [m.fget for m in find_members(Image, member_type=property)])
+def test_image_properties_have_docstrings(member):
+    assert member.__doc__ is not None
+
+# SIMPLE PROPERTIES TODO to be completed.
+
+
+@pytest.mark.parametrize("property_name,expected_path", [
+    ("directory", "frameInfo.directory"),
+    ("width", "frameInfo.fileInfoExtended.width"),
+])
+def test_simple_properties(image, property_name, expected_path, mock_get_value):
+    getattr(image, property_name)
+    mock_get_value.assert_called_with(expected_path)
+
+# TODO tests for all existing functions to be filled in
+
+
+def test_make_active(image, mock_session_call_action):
+    image.make_active()
+    mock_session_call_action.assert_called_with("setActiveFrameById", 0)
+
+
+@pytest.mark.parametrize("channel", [0, 10, 19])
+def test_set_channel_valid(image, channel, mock_call_action, mock_property):
+    mock_property("depth", 20)
+
+    image.set_channel(channel)
+    mock_call_action.assert_called_with("setChannels", channel, image.macro("", "requiredStokes"), True)
+
+
+@pytest.mark.parametrize("channel,error_contains", [
+    (20, "must be smaller"),
+    (1.5, "not an increment of 1"),
+    (-3, "must be greater or equal"),
+])
+def test_set_channel_invalid(image, channel, error_contains, mock_property):
+    mock_property("depth", 20)
+
+    with pytest.raises(CartaValidationFailed) as e:
+        image.set_channel(channel)
+    assert error_contains in str(e.value)
+
+
+@pytest.mark.parametrize("x", [0, 10, 19])
+@pytest.mark.parametrize("y", [0, 10, 19])
+def test_set_center_valid_pixels(image, mock_property, mock_call_action, x, y):
+    mock_property("width", 20)
+    mock_property("height", 20)
+
+    image.set_center(f"{x}px", f"{y}px")
+    mock_call_action.assert_called_with("setCenter", float(x), float(y))
+
+
+@pytest.mark.parametrize("x,y,x_fmt,y_fmt,x_norm,y_norm", [
```

Comment:
might be good to add test cases like (30d, -30d) where the latitude coordinate is a negative number. Need unit tests for this case too.

Reply:
This would probably be in the unit test for `set_center`.